src/Stratis.Bitcoin/Controllers/NodeController.cs

```diff
@@ -103,17 +144,257 @@ public IActionResult Status()
         }
 
         /// <summary>
-        /// Trigger a shoutdown of the current running node.
+        /// Gets the block header of the block identified by the hash.
+        /// API implementation of RPC call.
+        /// </summary>
+        /// <param name="hash">The block hash.</param>
+        /// <param name="isJsonFormat"><c>True to return Json formatted block header.</c></param>
+        /// <returns>Json formatted <see cref="BlockHeaderModel"/>. <c>null</c> if block not found. Returns <see cref="IActionResult"/> formatted error if fails.</returns>
+        /// <exception cref="NotImplementedException">Thrown if isJsonFormat = false</exception>"
+        /// <exception cref="ArgumentException">Thrown if hash is empty.</exception>
+        /// <exception cref="ArgumentNullException">Thrown if logger is not provided.</exception>
+        /// <remarks>Binary serialization is not supported with this method.</remarks>
+        [Route("getblockheader")]
+        [HttpGet]
+        public IActionResult GetBlockHeader([FromQuery] string hash, bool isJsonFormat = true)
+        {
+            try
+            {
+                Guard.NotNull(this.logger, nameof(this.logger));
+                Guard.NotEmpty(hash, nameof(hash));
+
+                this.logger.LogDebug("GetBlockHeader {0}", hash);
+                if (!isJsonFormat)
+                {
+                    this.logger.LogError("Binary serialization is not supported'{0}'.", nameof(GetBlockHeader));
```

Comment:
No need to add nameof the method that logged the error- it will be shown in the log anyway.